activation/interface.go

```diff
@@ -117,27 +120,61 @@ type SmeshingProvider interface {
 	SetCoinbase(coinbase types.Address)
 }
 
-// poetClient servers as an interface to communicate with a PoET server.
+type PoetAuth struct {
+	*PoetPoW
+	*certifier.PoetCert
+}
+
+// PoetClient servers as an interface to communicate with a PoET server.
 // It is used to submit challenges and fetch proofs.
-type poetClient interface {
+type PoetClient interface {
 	Address() string
 
+	// FIXME: remove support for deprecated poet PoW
 	PowParams(ctx context.Context) (*PoetPowParams, error)
 
+	// FIXME: remove support for deprecated poet PoW
 	// Submit registers a challenge in the proving service current open round.
 	Submit(
 		ctx context.Context,
 		deadline time.Time,
 		prefix, challenge []byte,
 		signature types.EdSignature,
 		nodeID types.NodeID,
-		pow PoetPoW,
+		auth PoetAuth,
 	) (*types.PoetRound, error)
 
+	CertifierInfo(context.Context) (*url.URL, []byte, error)
+
 	// Proof returns the proof for the given round ID.
 	Proof(ctx context.Context, roundID string) (*types.PoetProofMessage, []types.Member, error)
 }
 
+// A certifier client that the certifierService uses to obtain certificates
+// The implementation can use any method to obtain the certificate,
+// for example, POST verification.
+type certifierClient interface {
+	// The ID for which this client certifies.
+	Id() types.NodeID
+	// Certify the ID in the given certifier.
+	Certify(ctx context.Context, url *url.URL, pubkey []byte) (*certifier.PoetCert, error)
+}
+
+// certifierService is used to certify nodeID for registerting in the poet.
+// It holds the certificates and can recertify if needed.
+type certifierService interface {
+	// Acquire a certificate for the given poet.
+	GetCertificate(poet string) *certifier.PoetCert
```

Comment:
NIT: avoid redundant prefixes for methods

```suggestion
	Certificate(poet string) *certifier.PoetCert
```